### PR TITLE
Fix aside menu

### PR DESCRIPTION
**Pull request by KamiKillertO**
There's currently an issue with the aside menu button in the live demo app.
image

When we click on it, we are redirected to the documentation home page.
With this fix the aside menu button will now works as expected, when clicking on it it will open the side menu
image

I've also added a focus trap when the menu is opened

### CODE REVIEW

**Comment by lukemelia** [comment]
Hi @KamiKillertO Thanks for working on this. It looks like linting is failing on this PR. Can you fix that?

**Comment by KamiKillertO** [comment]
Sorry for that 😅

**Comment by KamiKillertO** [comment]
@lukemelia Done 👍

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "ember-cli-terser": "^4.0.2",
     "ember-code-snippet": "^3.0.0",
     "ember-export-application-global": "^2.0.1",
+    "ember-focus-trap": "^1.0.1",
```
Comment (bertdeblock) [inline package.json:70]: This has to go in `dependencies`.

**Review by bertdeblock — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (KamiKillertO) [reply package.json]: Done 👍

**Review by bertdeblock — APPROVED** [review]

**Comment by bertdeblock** [comment]
Seems there's still a conflict. Probably because I merged https://github.com/chrislopresto/ember-freestyle/pull/836, sorry for that. Looks good to go otherwise!

**Comment by bertdeblock** [comment]
We're running into https://github.com/embroider-build/embroider/issues/1175 because of the `ember-focus-trap` addition. This isn't an issue with `ember-freestyle` itself, but just wanted to point out that this change might break people's Embroider based builds.

**Comment by lukemelia** [comment]
@bertdeblock What is the ideal resolution from your perspective?

**Comment by bertdeblock** [comment]
Other than fixing https://github.com/embroider-build/embroider/issues/1175, I'm not sure to be honest.
The current workaround seems to be for the host app to install `ember-focus-trap` as well.
We could create an issue in `ember-freestyle` explaining this or even have this documented in the README or something until the bug is fixed. The only downside I guess is, we recommend to exclude `ember-freestyle` for production builds. If the host app itself doesn't use `ember-focus-trap` they end up with unused code in production builds unless they exclude `ember-focus-trap` manually themselves.

**Comment by bertdeblock** [comment]
FYI, https://github.com/embroider-build/embroider/pull/1278 should fix this issue for us.

**Pull request merged**